model/validation.go

```diff
@@ -669,3 +661,89 @@ func ValidateDestinationPolicy(msg proto.Message) error {
 
 	return errs
 }
+
+func validateProxyAddress(hostAddr string) error {
```

Comment:
Yup. That's why I reversed the check to be anything *not* ipv6. The documentation says checking against hostname isn't recommending since DNS can resolve to more than one address.